modules/app_service_private/main.tf

```diff
@@ -36,6 +36,20 @@ resource "azurerm_app_service" "app_service" {
   identity {
     type = "SystemAssigned"
   }
+
+  // Configures a mountpoint to a storage container (ro) or file share (rw)
+  // Read more: https://docs.microsoft.com/en-us/azure/app-service/configure-connect-to-azure-storage
+  dynamic "storage_account" {
+    for_each = var.storage_mount == null ? [] : [1]
```

Comment:
Could this have been done with `count` instead of `for_each`?

Reply:
Count can never be used in a dynamic block :) Strange but true..